### PR TITLE
Remove code that removed focus visual in SelectionFollowsFocus mode

### LINKED ISSUE #1539: Setting SelectionFollowsFocus causes focus rect to not render
**Describe the bug**

When NavigationView's SelectionFollowsFocus="Enabled", the menu items no longer render focus rects while they have keyboard focus.

**Steps to reproduce the bug**

1. Create new test app with 
```xaml
    <muxc:NavigationView PaneDisplayMode="Auto" SelectionFollowsFocus="Enabled">
        <muxc:NavigationView.MenuItems>
            <muxc:NavigationViewItem Icon="Play" Content="Menu Item1" Tag="SamplePage1" />
            <muxc:NavigationViewItem Icon="Save" Content="Menu Item2" Tag="SamplePage2" />
        </muxc:NavigationView.MenuItems>
        <Frame x:Name="contentFrame">
            <Grid>
                <Button>Hello</Button>
            </Grid>
        </Frame>
    </muxc:NavigationView>
```
2. Tab through the controls in the app
3. Note that the focus rect disappears between  hamburger and settings buttons.
4. Change SelectionFollowsFocus to Disabled.
5. Note focus draws as expected.

**Expected behavior**

When one of the MenuItems has keyboard focus, a focus rect is drawn around it.

**Screenshots**
NavigationViewFocus

**Additional context**

Copied from [internal bug](https://microsoft.visualstudio.com/OS/_workitems/edit/23886593)

### PR DESCRIPTION
**Pull request by chingucoding**
## Description

When SelectionFollowsFocus was set to enabled we disabled the focus visual, which isn't desired anymore.
## Motivation and Context

Fixes #1539 
## How Has This Been Tested?

Tested manually:
gif

### CODE REVIEW

File: dev/NavigationView/NavigationView.h [hunk 1]
```diff
@@ -165,7 +165,6 @@ class NavigationView :
     void UpdateVisualState(bool useTransitions = false);
     void UpdateVisualStateForOverflowButton();
     void UpdateLeftNavigationOnlyVisualState(bool useTransitions);
-    void UpdateNavigationViewUseSystemVisual();
     static void PropagateShowFocusVisualToAllNavigationViewItemsInRepeater(winrt::ItemsRepeater const& ir, bool showFocusVisual);
```
Comment (StephenLPeters) [inline dev/NavigationView/NavigationView.h:168]: >PropagateShowFocusVisualToAllNavigationViewItemsInRepeater [](start = 16, length = 58)

I think this isn't called anymore?

File: dev/NavigationView/NavigationView.xaml [hunk 2]
```diff
@@ -508,10 +508,6 @@
         </Setter>
     </Style>
 
-    <!--Note: Even though UseSystemFocusVisuals is set to True here in the Style, and because
-    bindings in Styles are only evaluated once, we were unable to bring a TemplateSettings.UseSystemFocusVisuals
```
Comment (StephenLPeters) [inline dev/NavigationView/NavigationView.xaml:512]: >bindings in Styles are only evaluated once [](start = 4, length = 42)

@Chingucoding and @Felix-Dev does this tid bit explain some of the Binding issues you've been seeing in other PRs?

Comment (chingucoding) [reply dev/NavigationView/NavigationView.xaml:512]: That certainly explains the issues I have seen. So the workaround is to essentially handle the binding inside the control logic then?

File: dev/NavigationView/NavigationView.h (hunk 1 above)
Comment (chingucoding) [reply dev/NavigationView/NavigationView.h]: Yes, this isn't used anymore, missed that function. Is removed now.

**Comment by StephenLPeters** [comment]
@chingucoding The pipeline fix just went in, could you please merge master into this PR?

**Comment by StephenLPeters** [comment]
@chingucoding I think this one was missed?

**Comment by chingucoding** [comment]
5f20943 was the merge with master which funnily came in before your message to merge with master.

**Review by StephenLPeters — APPROVED** [review]
:shipit:

**Pull request merged**